src/diffusers/models/unet_2d_blocks.py

```diff
@@ -460,7 +466,10 @@ def forward(self, hidden_states, temb=None):
         hidden_states = self.resnets[0](hidden_states, temb)
         for attn, resnet in zip(self.attentions, self.resnets[1:]):
             if attn is not None:
-                hidden_states = attn(hidden_states)
+                if self.use_spatial_norm:
```

Comment:
do we need this condition? `temb` is now an optional argument, we can pass it down as `None` when we do not use spatial_norm right? 

so basically this should be sufficient? 
```
if attn is not None:
   hidden_states = attn(hidden_states, temb)
```

Reply:
We cannot make changes about the spatial_norm at this point -> let's solve this like @yiyixuxu said in the attention / resnet